swagger/swagger.yaml

```diff
@@ -0,0 +1,299 @@
+- name: "member"
+  description: "Everything relating to church members interacting with the system"
+schemes:
+- "https"
+paths:
+  /user:
+    post:
+      tags:
+      - "member"
+      - "leader"
+      summary: "Create new user account"
+      description: "Create a new user to be authenticated via Firebase. Both church leaders and members need such an account."
+      operationId: "createUser"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "User details for new account"
+        required: true
+        schema:
+          $ref: "#/definitions/UserIn"
+      responses:
+        200:
+          description: "User account created"
+          schema:
+            $ref: "#/definitions/User"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorResponse"
+  /user/{userUuid}:
+    get:
+      tags:
+      - "member"
+      - "leader"
+      summary: "Retrieve user details"
+      description: "Retrieve user details including the user's contact details."
+      operationId: "getUser"
+      parameters:
+      - in: "path"
+        name: "userUuid"
+        type: "string"
+        required: true
+      responses:
+        200:
+          description: "User details"
+          schema:
+            $ref: "#/definitions/User"
+        404:
+          description: "User not found"
+          schema:
+            $ref:  "#/definitions/ErrorResponse"
+    put:
+      tags:
+      - "member"
+      - "leader"
+      summary: "Modify existing user"
+      description: "Edit user identified by user's UUID. All attributes can be edited."
+      operationId: "editUser"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "path"
+        name: "userUuid"
+        type: "string"
+        required: true
+      - in: "body"
+        name: "body"
+        description: "New user details"
+        required: true
+        schema:
+          $ref: "#/definitions/UserIn"
+      responses:
+        200:
+          description: "User account modified"
+          schema:
+            $ref: "#/definitions/User"
+        405:
+          description: "Invalid input"
+          schema:
+            $ref: "#/definitions/ErrorResponse"
+    delete:
+      tags:
+      - "member"
+      - "leader"
+      summary: "Delete user"
+      description: "Delete identified by user's UUID."
+      operationId: "deleteUser"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "path"
+        name: "userUuid"
+        type: "string"
+        required: true
+      responses:
+        204:
+          description: "User deleted"
+        404:
+          description: "User not found"
+          schema:
+            $ref: "#/definitions/ErrorResponse"
+
+  /church:
+    post:
+      tags:
+      - "member"
```

Comment:
members should not be allowed to add/remove/modify churches right? I can update this on my commit